iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -826,6 +834,18 @@ private Type.PrimitiveType getPrimitiveTypeOrThrow(FieldSchema field) throws Met
     return (Type.PrimitiveType) newType;
   }
 
+  private void setOrcOnlyFilesParam(org.apache.hadoop.hive.metastore.api.Table hmsTable) {
+    if (org.apache.iceberg.FileFormat.ORC.name().equalsIgnoreCase(hmsTable.getSd().getInputFormat()) ||
```

Comment:
nit: what about refactoring this huge condition to a method? (it can be even reused later)